marketplace-client/src/marketplace/src/app/_services/product-comparison/page-product-comparison.service.ts

```diff
@@ -0,0 +1,32 @@
+import { Injectable } from '@angular/core';
+import {Product} from "../../_models/products/product";
+import {ProductComparisonService} from "./product-comparison";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PageProductComparisonService implements ProductComparisonService{
+  private products: Product[] = []
+  constructor() {}
+
+  addProduct(product: Product): void {
+    if(this.products.indexOf(product) == -1){
+      this.products.push(product);
+    }
+  }
+
+  setProducts(products: Product[]){
+    this.products.splice(0);
+    products.forEach((product)=>{
+      this.addProduct(product);
+    },this)
```

Comment:
why's logic so complex? think about the number of operations performed...
Just copy 1 array in another, it's 1 line, 1 operation

https://developer.mozilla.org/ru/docs/Web/JavaScript/Reference/Operators/Spread_syntax